src/components/loadable-item.cjsx

```diff
@@ -28,57 +27,71 @@ module.exports = React.createClass
   getDefaultProps: ->
     bindEvent: 'change'
 
-  componentDidMount: -> @reload({})
-  componentDidUpdate: (oldProps) -> @reload(oldProps)
+  componentWillMount: ->
+    {id, store, options} = @props
+    @load(id, options) if id? and not store.isNew(id, options)
 
-  reload: (oldProps) ->
-    {id, store, load, actions, options} = @props
+  componentWillReceiveProps: (nextProps) ->
+    @reload(@props, nextProps)
+
+  arePropsSame: (prevProps, nextProps) ->
+    {id, store, load, actions, options} = nextProps
+
+    propsToCheck = ['id', 'store', 'load', 'actions', 'options']
+    _.isEqual(_.pick(prevProps, propsToCheck), _.pick(nextProps, propsToCheck))
+
```

Comment:
I like - that's a really nice and concise way to check only the props you care about.